grease-cli/grease-cli.cabal

```diff
@@ -0,0 +1,223 @@
+    -Werror=noncanonical-monoid-instances
+    -Werror=operator-whitespace
+    -Werror=operator-whitespace-ext-conflict
+    -Werror=orphans
+    -Werror=overflowed-literals
+    -Werror=overlapping-patterns
+    -Werror=partial-fields
+    -Werror=partial-type-signatures
+    -Werror=redundant-bang-patterns
+    -Werror=redundant-record-wildcards
+    -Werror=redundant-strictness-flags
+    -Werror=simplifiable-class-constraints
+    -Werror=star-binder
+    -Werror=star-is-type
+    -Werror=tabs
+    -Werror=type-defaults
+    -Werror=typed-holes
+    -Werror=type-equality-out-of-scope
+    -Werror=type-equality-requires-operators
+    -Werror=unicode-bidirectional-format-characters
+    -Werror=unrecognised-pragmas
+    -Werror=unrecognised-warning-flags
+    -Werror=unsupported-calling-conventions
+    -Werror=unsupported-llvm-version
+    -Werror=unticked-promoted-constructors
+    -Werror=unused-do-bind
+    -Werror=unused-imports
+    -Werror=unused-record-wildcards
+    -Werror=warnings-deprecations
+    -Werror=wrong-do-bind
+
+  if impl(ghc < 9.8)
+    ghc-options:
+      -Werror=forall-identifier
+
+  if impl(ghc >= 9.8)
+    ghc-options:
+      -Werror=incomplete-export-warnings
+      -Werror=inconsistent-flags
+      -Werror=missing-poly-kind-signatures
+
+  -- TODO(#44): Enable these warnings when supporting these GHC versions.
+  --
+  -- if impl(ghc >= 9.10)
+  --   -Werror=badly-staged-types
+  --   -Werror=data-kinds-tc
+  --   -Werror=deprecated-type-abstractions
+  --   -Werror=incomplete-record-selectors
+
+  -- if impl(ghc >= 9.12)
+  --   -Werror=view-pattern-signatures
+
+library
+  import: shared
+  hs-source-dirs: src
+  build-depends:
+    , grease
+    , aeson >= 2
+    , async
+    , bytestring
+    , containers
+    , directory
+    , exceptions
+    , file-embed
+    , filepath
+    , gitrev
+    , lens
+    , libBF
+    , lumberjack >= 1.0 && < 1.1
+    , megaparsec
+    , mtl
+    , optparse-applicative
+    , panic >= 0.3
+    , safe-exceptions
+    , text
+    , time
+    , transformers
+    , vector
+
+    , prettyprinter
+    , parameterized-utils
+    , bv-sized
+    , what4
+    , llvm-pretty
+    , llvm-pretty-bc-parser
+    , crucible
+    , crucible-debug
+    , crucible-llvm-debug
+    , crucible-llvm
+    , crucible-llvm-syntax
+    , crucible-macaw-debug
+    , crucible-syntax
+    , elf-edit
+    , elf-edit-core-dump
+    , asl-translator
+    , macaw-loader
+    , macaw-loader-aarch32
+    , macaw-loader-ppc
+    , macaw-loader-x86
+    , macaw-base
+    , macaw-symbolic
+    , macaw-symbolic-syntax
+    , macaw-aarch32
+    , macaw-aarch32-symbolic
+    , macaw-aarch32-syntax
+    , macaw-ppc
+    , macaw-ppc-symbolic
+    , macaw-ppc-syntax
+    , macaw-x86
+    , macaw-x86-symbolic
+    , macaw-x86-syntax
+    , stubs-common
+    , stubs-wrapper
+
+  exposed-modules:
+    Grease.Main
+    Grease.Profiler.EmbeddedData
+    Grease.Profiler.Feature
+    Grease.Profiler.Paths
```

Comment:
Should these perhaps be `other-modules`? Are they depended on by the test suite?

Reply:
No, they aren't imported anywhere except `Grease.Main`. The only reason that I made them exposed modules was because that is how `grease` (before the package split) was treating them. There's no reason we have to do the same now that we've moved them to `grease-cli`, of course.